Update base_sha and head_sha for kernel checkers

hasiburr-qti opened the pull request:
# Description

This PR updates the sync action and ensures that the `base_sha`, `head_sha` and the `kernel_src` variables are properly set after the kernel repo is checked out and the workspace is ready for the checkers to run.
Later these values are passed on to the checker script to ensure the actual changes are detected by the check run to analyze the changes in the PR.

hasiburr-qti committed:
Update base_sha and head_sha for kernel checkers

This commit updates the sync action and ensures that the base_sha head_sha and the kernel_src variables are properly set after the kernel repo is checked out and the workspace is ready for the checkers to run.

Signed-off-by: Hasibur Rahman Mohammed <hasiburr@qti.qualcomm.com>

jasonpettit commented:
Can this PR be closed?

sgaud-quic commented on .github/actions/sync/action.yml:
```diff
@@ -103,7 +102,9 @@ runs:
         else
```
this is not correct for qcom-6.18.y branch?

sgaud-quic commented on .github/actions/sync/action.yml:
```diff
@@ -37,57 +41,106 @@ runs:
-        if ! git diff --cached --quiet; then
-          git commit -m "Merged PR #${{ inputs.pr_number }}"
+
+        if [ "$BASE" = "qcom-next-staging" ]; then
```
Instead of comparing branch, you can compare repo : 
if [ "$TOPIC_REPO" = "qualcomm-linux/kernel" ]; then

This will handle condition for all branches in kernel repository

hasiburr-qti replied:
Done

hasiburr-qti committed:
Update base_sha and head_sha for kernel checkers

This commit updates the sync action and ensures that the base_sha
head_sha and the kernel_src variables are properly set after the
kernel repo is checked out and the workspace is ready for the checkers to run.

Signed-off-by: Hasibur Rahman Mohammed <hasiburr@qti.qualcomm.com>
